satpy/readers/fci_l2_nc.py

```diff
@@ -61,54 +107,28 @@ def __init__(self, filename, filename_info, filetype_info):
 
     @property
     def start_time(self):
-        """Get observation start time."""
-        try:
-            start_time = datetime.strptime(self.nc.attrs['time_coverage_start'], '%Y%m%d%H%M%S')
-        except (ValueError, KeyError):
-            # TODO if the sensing_start_time_utc attribute is not valid, uses a hardcoded value
-            logger.warning("Start time cannot be obtained from file content, using default value instead")
-            start_time = datetime.strptime('20200101120000', '%Y%m%d%H%M%S')
-        return start_time
+        return CommonFunctions.get_start_time(self.nc.attrs['time_coverage_start'])
 
     @property
     def end_time(self):
-        """Get observation end time."""
-        try:
-            end_time = datetime.strptime(self.nc.attrs['time_coverage_end'], '%Y%m%d%H%M%S')
-        except (ValueError, KeyError):
-            # TODO if the sensing_end_time_utc attribute is not valid, adds 20 minutes to the start time
-            end_time = self.start_time + timedelta(minutes=PRODUCT_DATA_DURATION_MINUTES)
-        return end_time
+        return CommonFunctions.get_end_time(self.nc.attrs['time_coverage_end'], self.start_time)
 
     @property
     def spacecraft_name(self):
-        """Return spacecraft name."""
-        try:
-            return self.nc.attrs['platform']
-        except KeyError:
-            # TODO if the platform attribute is not valid, return a default value
-            logger.warning("Spacecraft name cannot be obtained from file content, using default value instead")
-            return 'DEFAULT_MTG'
+        return CommonFunctions.get_spacecraft_name(self.nc.attrs)
 
     @property
     def sensor(self):
-        """Return instrument."""
-        try:
-            return self.nc.attrs['data_source']
-        except KeyError:
-            # TODO if the data_source attribute is not valid, return a default value
-            logger.warning("Sensor cannot be obtained from file content, using default value instead")
-            return 'fci'
+        return CommonFunctions.get_sensor_name(self.nc.attrs)
```

Comment:
Looks like these method are exactly the same across the different file handler. Why not make a base class for it and use inheritance?

Reply:
ok, i think i have done this the way you have suggested

Reply:
almost! I was actually meaning that you could have the property in the base class too.

Reply:
how does it look now?

Reply:
@mraspaud - did you have any comment on my changes to resolve this conversation?

Reply:
sorry for not getting back to you earlier! yes that works for me!